tensorflow_recommenders_addons/dynamic_embedding/python/ops/hkv_hashtable_ops.py

```diff
@@ -0,0 +1,517 @@
+              self.resource_handle,
+              keys,
+              dynamic_default_values
+              if dynamic_default_values is not None else self._default_value,
+          )
+    return (values, exists) if return_exists else values
+
+  def insert(self, keys, values, name=None, shared_name=None):
+    """Associates `keys` with `values`.
+
+        Args:
+          keys: Keys to insert. Can be a tensor of any shape. Must match the table's
+            key type.
+          values: Values to be associated with keys. Must be a tensor of the same
+            shape as `keys` and match the table's value type.
+          name: A name for the operation (optional).
+
+        Returns:
+          The created Operation.
+
+        Raises:
+          TypeError: when `keys` or `values` doesn't match the table data
+            types.
+        """
+    with ops.name_scope(
+        name,
+        "%s_lookup_table_insert" % self.name,
+        [self.resource_handle, keys, values],
+    ):
+      keys = ops.convert_to_tensor(keys, self._key_dtype, name="keys")
+      values = ops.convert_to_tensor(values, self._value_dtype, name="values")
+      with ops.colocate_with(self.resource_handle, ignore_existing=True):
+        # pylint: disable=protected-access
+        op = hkv_ops.tfra_hkv_hash_table_insert(self.resource_handle, keys,
+                                                values)
+    return op
+
+  def accum(self, keys, values_or_deltas, exists, name=None, shared_name=None):
+    """Associates `keys` with `values`.
+
+      Args:
+        keys: Keys to accmulate. Can be a tensor of any shape.
+          Must match the table's key type.
+        values_or_deltas: values to be associated with keys. Must be a tensor of
+          the same shape as `keys` and match the table's value type.
+        exists: A bool type tensor indicates if keys already exist or not.
+          Must be a tensor of the same shape as `keys`.
+        name: A name for the operation (optional).
+
+      Returns:
+        The created Operation.
+
+      Raises:
+        TypeError: when `keys` or `values` doesn't match the table data
+          types.
+    """
+    with ops.name_scope(
+        name,
+        "%s_lookup_table_accum" % self.name,
+        [self.resource_handle, keys, values_or_deltas],
+    ):
+      keys = ops.convert_to_tensor(keys, self._key_dtype, name="keys")
+      values_or_deltas = ops.convert_to_tensor(values_or_deltas,
+                                               self._value_dtype,
+                                               name="values_or_deltas")
+      exists = ops.convert_to_tensor(exists, dtypes.bool, name="exists")
+      with ops.colocate_with(self.resource_handle, ignore_existing=True):
+        # pylint: disable=protected-access
+        op = hkv_ops.tfra_hkv_hash_table_accum(self.resource_handle, keys,
+                                               values_or_deltas, exists)
+    return op
+
+  def export(self, name=None):
```

Comment:
What will happen if the size of the table is larger than available HBM?

Reply:
The OOM error will be reported, but because the bucket was not optimized before, it will leave a lot of logs, resulting in a dead pod. When I didn't find out the specific cause, I thought it was a dead loop. Now after optimization, OOM error will be reported here